druid/src/main/java/org/apache/calcite/adapter/druid/TimeExtractionFunction.java

```diff
@@ -34,6 +41,12 @@
  */
 public class TimeExtractionFunction implements ExtractionFunction {
 
+  public static final List<TimeUnitRange> LIST_OF_VALID_TIME_EXTRACT = ImmutableList.of(
```

Comment:
I think this can be private. You could use [Sets.immutableEnumSet](http://google.github.io/guava/releases/snapshot/api/docs/com/google/common/collect/Sets.html#immutableEnumSet-java.lang.Iterable-) rather than a List.